### DISCUSSION TITLE
Cascade lookup on current logged in user id?

### DISCUSSION THREAD

**Discussion by rchawla-alphabyte** [serenity-is/Serenity#5741] [Help]
Hello.

I am trying to filter a lookup in a row.cs field using the cascadefield/cascadevalue attribute.

Would there be anyway to retrieve the current logged in user as the 'CascadeValue' attribute within the row.cs file or would the filtering have to be done on the client side? 

Any suggestions would be appreciated!

![Lookup](https://user-images.githubusercontent.com/76563021/119500381-bc008880-bd35-11eb-8863-688174b89403.png)

**Comment by minhhungit** [comment]
There is a solution I posted here https://github.com/serenity-is/Serenity/issues/4960#issuecomment-611901231
Instead ApplyCurrentLanguageFilter , you can update it to apply current userId, ex:
```csharp
public static SqlQuery ApplyCurrentUserId(this SqlQuery query)
{
    // load userId
    var userId = Serenity.Authorization.UserId; // for example
    query.Select($"{userId}", "UserId");
    return query;
}
```
You can get userId using `Serenity.Authorization.UserId` to get userId (if you are using .net 5 you need to use `userAccessor` or something like that)
The purpose of above code is adding UserId field into your List/Retrieve query which serenity generated

After you have UserId field in your List/Retrieve query, you can add an Expression field for example :

```csharp
[DisplayName("UserId"), Expression("bl.UserId")]
public String UserId
{
    get { return Fields.UserId[this]; }
    set { Fields.UserId[this] = value; }
}
```

Finally you can use the userId field for your lookup cascade 

I have not **re-test** it for your case but I think it will work, you might need to write more code if you need

**Reply by minhhungit** [reply]
Instead of using above solution, you can use Serenity Behavior to modify list/retrieve query, this way is cleaner

**Reply by wirble** [reply]
Can you elaborate? I am not aware there is a built in Serenity behavior that filters by ID. in the past I usually modify the repository to filter by ID.  Is there a better way?

**Reply by minhhungit** [reply]
@wirble read serenity document, in multi-tenant demo, there is a sample about behavior 
https://serenity.is/docs/tutorials/multi_tenancy/using_serenity_service_behaviors

**Reply by wirble** [reply]
Thanks. I read the manual before. Must have missed this one or skipped it since I didn’t use tenancy.

**Reply by wirble** [reply]
In your #4960 when you create a custom handler. How do you get Serenity to use the new customer handler instead of the default generated handler?

**Reply by minhhungit** [reply]
Registry `IMultiTenantRow` into your row.cs, follow document to see how it works
